cmd/client/main/main.go

```diff
@@ -101,7 +101,33 @@ func main() {
 	}
 
 	if *fl.updateInfo {
-		// TODO: Send a update info message, receive the response and update the wallet balance/nonce
+		fmt.Println("Checking for new wallet info...")
+		accountInfo, err := client.RequestWalletInfo(*fl.producer)
```

Comment:
Can you first check if `*fl.producer` exists? If it doesn't, return a `Fatal` error stating that it's required.